cabal-install-solver/src/Distribution/Solver/Types/PkgConfigDb.hs

```diff
@@ -63,10 +68,14 @@ readPkgConfigDb verbosity progdb = handle ioErrorHandler $ do
     case mpkgConfig of
       Nothing             -> noPkgConfig "Cannot find pkg-config program"
       Just (pkgConfig, _) -> do
-        pkgList <- lines <$> getProgramOutput verbosity pkgConfig ["--list-all"]
+        -- To prevent malformed Unicode in the descriptions from crashing cabal,
+        -- read without interpreting any encoding first. (#9608)
+        pkgList <- LBS.split 10 <$> getProgramInvocationLBS verbosity (programInvocation pkgConfig ["--list-all"])
         -- The output of @pkg-config --list-all@ also includes a description
         -- for each package, which we do not need.
-        let pkgNames = map (takeWhile (not . isSpace)) pkgList
+        let pkgNamesLBS = map (LBS.takeWhile (not . isSpace . chr . fromIntegral)) pkgList
```

Comment:
It will not! You are correct, `isSpace` also recognises non-breaking spaces (U+00A0) which, when considered mod 256, have the high bit set and can thus occur in UTF-8 text as part of a multi-byte unicode codepoint. I had naively expected there to be no space characters in the Unicode range between 127 and 256, which would have made this work due to the UTF-8 encoding.

What do you suggest? Something like `not . isAsciiSpace` where ``isAsciiSpace n = n `elem` map (fromIntegral . ord) " \t"``?